replica/tablets.cc

```diff
@@ -200,6 +204,14 @@ future<tablet_metadata> read_tablet_metadata(cql3::query_processor& qp) {
             current = active_tablet_map{table, tmap, tmap.first_tablet()};
         }
 
+        if (row.has("resize")) {
+            auto resize_decision_name = row.get_as<sstring>("resize");
+            api::timestamp_type resize_generation = row.get_as<int64_t>("resize_generation");
```

Comment:
timestamp_type? is it a timestamp or a generation?

what if it's null?

Reply:
I decouple the type from the generation. generation is a concept, for global identification, that can have any type. think of sstable generation, we moved from integer type to uuid. timestamp is a possible type too.

as for it being null, we should never allow that to happen in practice. abort or construct resize metadata with default generation?

Reply:
If we create the record with a generation, then ok.

But the type and name have to match. A reader that sees timestamp will assume it behaves as a timestamp. A reader that sees generation will assume it behaves like generation. We should randomly mix things, it will cause confusion.

Reply:
Yes, I am switching to seq number already. It will cause confusion that timestamp is used mainly for identification.